presto-main/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -0,0 +1,88 @@
+
+public class MaterializedViewUtils
+{
+    private MaterializedViewUtils() {}
+
+    public static void validateMaterializedViewQueryPlan(Statement query)
+    {
+        MaterializedViewPlanValidator validator = new MaterializedViewPlanValidator(query);
+        validator.process(query, new MaterializedViewPlanValidatorContext());
+    }
+
+    public static Map<String, Map<SchemaTableName, String>> computeMaterializedViewToBaseTableColumnMappings(Query viewQuery, Analysis analysis)
+    {
+        ImmutableMap.Builder<String, Map<SchemaTableName, String>> fullColumnMapping = ImmutableMap.builder();
+
+        Map<String, Map<SchemaTableName, String>> originalColumnMapping = getOriginalColumnMappingFromAnalysis(analysis, viewQuery);
+
+        MaterializedViewPlanValidator planAnalyzer = new MaterializedViewPlanValidator(viewQuery, true, Optional.of(analysis));
+        planAnalyzer.process(viewQuery, new MaterializedViewPlanValidatorContext());
+        List<List<TableColumn>> linkedBaseColumns = planAnalyzer.getLinkedBaseColumns();
+
+        for (Map.Entry<String, Map<SchemaTableName, String>> entry : originalColumnMapping.entrySet()) {
+            String viewColumn = entry.getKey();
+            Map<SchemaTableName, String> originalBaseColumns = entry.getValue();
+
+            Map<SchemaTableName, String> fullBaseColumns = new HashMap<SchemaTableName, String>() {{
+                    putAll(originalBaseColumns); }};
+
+            originalBaseColumns.forEach((originalBaseTable, originalBaseColumn) -> {
+                TableColumn originalTableColumn = new TableColumn(originalBaseTable, originalBaseColumn);
+                linkedBaseColumns.forEach(linkedTableColumnPair -> {
+                    if (originalTableColumn.equals(linkedTableColumnPair.get(0))) {
+                        fullBaseColumns.put(linkedTableColumnPair.get(1).getTableName(), linkedTableColumnPair.get(1).getColumnName());
+                    }
+                    else if (originalTableColumn.equals(linkedTableColumnPair.get(1))) {
+                        fullBaseColumns.put(linkedTableColumnPair.get(0).getTableName(), linkedTableColumnPair.get(0).getColumnName());
+                    }
+                });
+            });
+
+            fullColumnMapping.put(viewColumn, ImmutableMap.copyOf(fullBaseColumns));
+        }
+
+        return fullColumnMapping.build();
+    }
+
+    private static Map<String, Map<SchemaTableName, String>> getOriginalColumnMappingFromAnalysis(Analysis analysis, Node node)
+    {
+        return analysis.getOutputDescriptor(node).getVisibleFields().stream()
+                .filter(field -> field.getOriginTable().isPresent() && field.getOriginColumnName().isPresent())
+                .collect(toImmutableMap(
+                        field -> field.getName().get(),
+                        field -> ImmutableMap.of(toSchemaTableName(field.getOriginTable().get()), field.getOriginColumnName().get())));
```

Comment:
Though it is illegal to have non-named field or original table, it would still be good to use `orElseThrow` than `get` in case the field name or table is missing.

Reply:
Before getting to this line, we have the filter to filter out the fields that don't have original table or original column. So it's guaranteed it won't throw here.